### PR TITLE
Add rounded corners to WaitTimesChart

### PR DESCRIPTION
**Pull request by connorreinhold**
## Overview

Added rounded edges to our WaitTimesComponent to be more in line with designs and with iOS (corner radius was set to 10)

## Changes Made

Created class BarChartRoundedRenderer, an extension of BarChartRenderer. Overrode methods of drawing rectangles to draw rounded rectangles (with a non-rounded rectangle below, so it would just be rounded on top). Wrote the drawRectHelper method to help with drawing this. In setupWaitTimesChart() method of WaitTimesComponent, set the renderer to be our customized BarChartRoundedRenderer.

## Test Coverage

Simulators & test device

## Screenshots
Before: 
Screen Shot 2019-11-06 at 12 20 09 PM

After:
image

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+                                            c,
+                                            icon,
+                                            (int)(x + iconsOffset.x),
+                                            (int)(y + iconsOffset.y),
+                                            icon.getIntrinsicWidth(),
+                                            icon.getIntrinsicHeight());
+                                }
+                            }
+                        }
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: Delete newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+                        bufferIndex = vals == null ? bufferIndex + 4 : bufferIndex + 4 * vals.length;
+                        index++;
+                    }
+                }
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: Delete newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+        c.drawText(valueText, x, y, mValuePaint);
+    }
+
+    @Override
+    public void drawHighlighted(Canvas c, Highlight[] indices) {
+
+        BarData barData = mChart.getBarData();
+
+        for (Highlight high : indices) {
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: delete newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+            IBarDataSet set = barData.getDataSetByIndex(high.getDataSetIndex());
+
+            if (set == null || !set.isHighlightEnabled())
+                continue;
+
+            BarEntry e = set.getEntryForXValue(high.getX(), high.getY());
+
+            if (!isInBoundsX(e, set))
+                continue;
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: Add { } for if statements

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+            mHighlightPaint.setColor(set.getHighLightColor());
+            mHighlightPaint.setAlpha(set.getHighLightAlpha());
+
+            boolean isStack = (high.getStackIndex() >= 0  && e.isStacked());
+
+            final float y1;
+            final float y2;
+
+            if (isStack) {
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 6]
```diff
@@ -0,0 +1,522 @@
...
+                if(mChart.isHighlightFullBarEnabled()) {
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 7]
```diff
@@ -0,0 +1,522 @@
...
+                    y1 = e.getPositiveSum();
+                    y2 = -e.getNegativeSum();
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+                } else {
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 9]
```diff
@@ -0,0 +1,522 @@
...
+                    Range range = e.getRanges()[high.getStackIndex()];
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 10]
```diff
@@ -0,0 +1,522 @@
...
+                    y1 = range.from;
+                    y2 = range.to;
+                }
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 11]
```diff
@@ -0,0 +1,522 @@
...
+            } else {
+                y1 = e.getY();
+                y2 = 0.f;
+            }
+
+            prepareBarHighlight(e.getX(), y1, y2, barData.getBarWidth() / 2f, trans);
+
+            setHighlightDrawPos(high, mBarRect);
+
+            drawRectHelper(c, mBarRect, mHighlightPaint);
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: can we del newlines and condense this

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 12]
```diff
@@ -0,0 +1,522 @@
...
+
+            if (drawBorder) {
+                drawRectHelper(c, buffer.buffer[j], buffer.buffer[j + 1], buffer.buffer[j + 2],
+                        buffer.buffer[j + 3], mBarBorderPaint);
+            }
+        }
+    }
+
+    private void prepareBarHighlight(float x, float y1, float y2, float barWidthHalf, Transformer trans) {
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 13]
```diff
@@ -0,0 +1,522 @@
...
+        trans.rectToPixelPhase(mBarRect, mAnimator.getPhaseY());
+    }
+
+    @Override
+    public void drawValues(Canvas c) {
+
+        // if values are drawn
+        if (isDrawingValuesAllowed(mChart)) {
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 14]
```diff
@@ -0,0 +1,522 @@
...
+            List<IBarDataSet> dataSets = mChart.getBarData().getDataSets();
+
+            final float valueOffsetPlus = Utils.convertDpToPixel(4.5f);
+            float posOffset = 0f;
+            float negOffset = 0f;
+            boolean drawValueAboveBar = mChart.isDrawValueAboveBarEnabled();
+
+            for (int i = 0; i < mChart.getBarData().getDataSetCount(); i++) {
+
```
Comment (YKo20010) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: del newline

**Comment by connorreinhold** [comment]
Updated looks like this -- now the bottom is flat but the top still isn't very curved. I think it looks better now
image

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 15]
```diff
@@ -0,0 +1,522 @@
...
+        if (mChart.isDrawBarShadowEnabled()) {
+            mShadowPaint.setColor(dataSet.getBarShadowColor());
+
+            BarData barData = mChart.getBarData();
+
+            final float barWidth = barData.getBarWidth();
+            final float barWidthHalf = barWidth / 2.0f;
+            float x;
+
+            for (int i = 0, count = Math.min((int)(Math.ceil((float)(dataSet.getEntryCount()) * phaseX)), dataSet.getEntryCount());
```
[the remark this answers is not in the document]
Comment (connorreinhold) [reply app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: Open-sourced guys thought they were funny. Changed it

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 16]
```diff
@@ -0,0 +1,522 @@
...
+    BarChartRoundedRenderer(BarDataProvider chart, ChartAnimator animator,
+            ViewPortHandler viewPortHandler) {
+        super(animator, viewPortHandler);
+        this.mChart = chart;
+
+        mHighlightPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
+        mHighlightPaint.setStyle(Paint.Style.FILL);
+        mHighlightPaint.setColor(Color.rgb(0, 0, 0));
+        // set alpha after color
+        mHighlightPaint.setAlpha(120);
```
[the remark this answers is not in the document]
Comment (connorreinhold) [reply app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: This is just when the graph is initialized - it updates the highlight color to the correct value whenever it draws the highlight with:

            mHighlightPaint.setColor(set.getHighLightColor());
            mHighlightPaint.setAlpha(set.getHighLightAlpha());

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 17]
```diff
@@ -0,0 +1,522 @@
...
+
+        BarData barData = mChart.getBarData();
+        mBarBuffers = new BarBuffer[barData.getDataSetCount()];
+
+        for (int i = 0; i < mBarBuffers.length; i++) {
+            IBarDataSet set = barData.getDataSetByIndex(i);
+            mBarBuffers[i] = new BarBuffer(set.getEntryCount() * 4 * (set.isStacked() ? set.getStackSize() : 1),
+                    barData.getDataSetCount(), set.isStacked());
+        }
+    }
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: add a quick comment about what this chunk of code does

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 18]
```diff
@@ -0,0 +1,427 @@
...
+        float phaseY = mAnimator.getPhaseY();
+        // draw the bar shadow before the values
+        if (mChart.isDrawBarShadowEnabled()) {
+            mShadowPaint.setColor(dataSet.getBarShadowColor());
+
+            BarData barData = mChart.getBarData();
+            final float barWidth = barData.getBarWidth();
+            final float barWidthHalf = barWidth / 2.0f;
+            float x;
+            int count = count = Math.min((int)(Math.ceil((float)(dataSet.getEntryCount()) * phaseX)), dataSet.getEntryCount());
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: one too many `count =`

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 19]
```diff
@@ -0,0 +1,183 @@
...
+            for (int i = 0; i < count; i++) {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: add a comment for this for loop

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 20]
```diff
@@ -0,0 +1,183 @@
...
+        buffer.setInverted(mChart.isInverted(dataSet.getAxisDependency()));
+        buffer.setBarWidth(mChart.getBarData().getBarWidth());
+
+        buffer.feed(dataSet);
+        trans.pointValuesToPixel(buffer.buffer);
+        final boolean isSingleColor = dataSet.getColors().size() == 1;
+        if (isSingleColor) {
+            mRenderPaint.setColor(dataSet.getColor());
+        }
+        for (int j = 0; j < buffer.size(); j += 4) {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: add a comment for this for loop

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 21]
```diff
@@ -0,0 +1,183 @@
...
+import com.github.mikephil.charting.interfaces.dataprovider.BarDataProvider;
+import com.github.mikephil.charting.interfaces.datasets.IBarDataSet;
+import com.github.mikephil.charting.renderer.BarChartRenderer;
+import com.github.mikephil.charting.utils.Transformer;
+import com.github.mikephil.charting.utils.Utils;
+import com.github.mikephil.charting.utils.ViewPortHandler;
+import com.github.mikephil.charting.model.GradientColor;
+import android.graphics.LinearGradient;
+
+public class BarChartRoundedRenderer extends BarChartRenderer {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: leave a comment here explaining the class and where to get more info (ie link the repo)

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java [hunk 22]
```diff
@@ -0,0 +1,183 @@
+package com.cornellappdev.android.eatery.components;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.RectF;
+
+import com.github.mikephil.charting.animation.ChartAnimator;
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: idk which ones you don't need, but make sure to delete any imports that you no longer need!

File: app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java (hunk 19 above)
Comment (connorreinhold) [reply app/src/main/java/com/cornellappdev/android/eatery/components/BarChartRoundedRenderer.java]: Deleted whole section we don't use shadows

**Pull request merged**